### PR TITLE
perf: use get_logits() to avoid massive GPU→CPU logits copy in ortgenai evaluator

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

Replace `get_output('logits')` with `get_logits()` in the ortgenai evaluator to avoid copying the full logits tensor from GPU to CPU each call.

## Problem

`generator.get_output('logits')` copies the **entire** logits tensor (`[batch, seq_len, vocab_size]`) from GPU to CPU. For a 900-token prompt with 262K vocab, this is **472MB per call**, taking **~410ms** — more than the forward pass itself (289ms).

For loglikelihood evaluation, we only need logits at the continuation token positions (typically 1–20 positions out of 900). Copying all 900 positions is wasteful.

## Fix

Use `generator.get_logits()` which returns only the last position's logits (~1MB, ~2ms). The evaluator now always uses incremental token appending: bulk-prefill the context, then step through continuation tokens collecting logits at each position.

## Benchmark

Gemma4 E2B-IT, MMLU Pro limit=50, CUDA EP, H200:

| Metric | Before | After |
|--------|--------|-------|
| Time | 46.9s | **24.2s** |
| Throughput | 10.6 req/s | **20.7 req/s** |
| **Speedup** | — | **1.94×** |
| Accuracy | 16.0% | 16.0% ✅ |

### Per-call impact

| API | Copy size | Time |
|-----|----------|------|
| `get_output('logits')` | 472MB (all positions) | ~410ms |
| `get_logits()` | ~1MB (last position) | ~2ms |

Full analysis: [onnxruntime/mobius#285](https://github.com/onnxruntime/mobius/issues/285)

### CODE REVIEW

File: olive/evaluator/lmeval_ort.py [hunk 1]
```diff
@@ -551,27 +551,25 @@ def model_call(self, input_ids: torch.Tensor, cont_len: int = 0) -> torch.Tensor
...
-            generator.append_tokens(input_ids.tolist())
-            return torch.from_numpy(generator.get_output("logits")).to(self.device)
-
-        # Model only returns logits for the last appended position.
         if batch_size > 1 and cont_len > 1:
             raise ValueError(
                 "batch_size > 1 is not supported when the model returns single-position logits"
                 " and continuation length > 1. Right-padding misaligns continuation positions across"
                 " batch elements. Use batch_size=1 instead."
             )
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply olive/evaluator/lmeval_ort.py:559]: @copilot apply changes based on [this feedback](https://github.com/microsoft/Olive/pull/2452#discussion_r3209998121)

**Review by jambayk — APPROVED** [review]
